clcache.py

```diff
@@ -864,7 +864,11 @@ def parseArgumentsAndInputFiles(cmdline):
                     else:
                         raise AssertionError("Unsupported argument type.")
 
-                    arguments[arg.name].append(value)
+                    # consistently use empty list for flags like /c or /doc
+                    if not arg.name in arguments:
+                        arguments[arg.name] = []
+                    if value:
+                        arguments[arg.name].append(value)
```

Comment:
With this change, the appending is only done when `value` is non-empty. But in both cases (empty/non-empty `value`), we need to set the dictionary key.